## Read and Clean Dataset


Import the libraries.


In [1]:
# Install tensorflow and keras in Anaconda
# !pip install tensorflow
# !pip install keras

In [2]:
import pandas as pd
import numpy as np
import keras
from keras.models import Sequential
from keras.layers import Dense
from sklearn.model_selection import train_test_split
from sklearn import metrics

<strong>The dataset is about the compressive strength of different samples of concrete based on the volumes of the different ingredients that were used to make them. Ingredients include:</strong>

<strong>1. Cement</strong>

<strong>2. Blast Furnace Slag</strong>

<strong>3. Fly Ash</strong>

<strong>4. Water</strong>

<strong>5. Superplasticizer</strong>

<strong>6. Coarse Aggregate</strong>

<strong>7. Fine Aggregate</strong>


Read the data into a <em>pandas</em> dataframe.


In [3]:
concrete_data = pd.read_csv('concrete_data.csv')
concrete_data.head()

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  Strength  
0            1040.0           676.0   28     79.99  
1            1055.0           676.0   28     61.89  
2             932.0           594.0  270     40.27  
3             932.0           594.0  365     41.05  
4             978.4           825.5  360     44.30

So the first concrete sample has 540 cubic meter of cement, 0 cubic meter of blast furnace slag, 0 cubic meter of fly ash, 162 cubic meter of water, 2.5 cubic meter of superplaticizer, 1040 cubic meter of coarse aggregate, 676 cubic meter of fine aggregate. Such a concrete mix which is 28 days old, has a compressive strength of 79.99 MPa.


#### Let's check how many data points we have.


In [4]:
concrete_data.shape

(1030, 9)

So, there are approximately 1000 samples to train our model on. Because of the few samples, we have to be careful not to overfit the training data.


Let's check the dataset for any missing values.


In [5]:
concrete_data.describe()

Cement  Blast Furnace Slag      Fly Ash        Water  \
count  1030.000000         1030.000000  1030.000000  1030.000000   
mean    281.167864           73.895825    54.188350   181.567282   
std     104.506364           86.279342    63.997004    21.354219   
min     102.000000            0.000000     0.000000   121.800000   
25%     192.375000            0.000000     0.000000   164.900000   
50%     272.900000           22.000000     0.000000   185.000000   
75%     350.000000          142.950000   118.300000   192.000000   
max     540.000000          359.400000   200.100000   247.000000   

       Superplasticizer  Coarse Aggregate  Fine Aggregate          Age  \
count       1030.000000       1030.000000     1030.000000  1030.000000   
mean           6.204660        972.918932      773.580485    45.662136   
std            5.973841         77.753954       80.175980    63.169912   
min            0.000000        801.000000      594.000000     1.000000   
25%            0.000000        932.000000      730.950000     7.000000   
50%            6.400000        968.000000      779.500000    28.000000   
75%           10.200000       1029.400000      824.000000    56.000000   
max           32.200000       1145.000000      992.600000   365.000000   

          Strength  
count  1030.000000  
mean     35.817961  
std      16.705742  
min       2.330000  
25%      23.710000  
50%      34.445000  
75%      46.135000  
max      82.600000

In [6]:
concrete_data.isnull().sum()

Cement                0
Blast Furnace Slag    0
Fly Ash               0
Water                 0
Superplasticizer      0
Coarse Aggregate      0
Fine Aggregate        0
Age                   0
Strength              0
dtype: int64

The data looks very clean and is ready to be used to build our model.


#### Split data into predictors (X) and target (y)


The target variable in this problem is the concrete sample strength. Therefore, our predictors will be all the other columns.


In [7]:
concrete_data_columns = concrete_data.columns

X = concrete_data[concrete_data_columns[concrete_data_columns != 'Strength']] # all columns except Strength
y = concrete_data['Strength'] # Strength column

<a id="item2"></a>


Let's do a quick sanity check of the predictors and the target dataframes.


In [8]:
X.head()

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  
0            1040.0           676.0   28  
1            1055.0           676.0   28  
2             932.0           594.0  270  
3             932.0           594.0  365  
4             978.4           825.5  360

In [9]:
y.head()

0    79.99
1    61.89
2    40.27
3    41.05
4    44.30
Name: Strength, dtype: float64

Let's save the number of predictors to *n_cols* since we will need this number when building our network.


In [10]:
n_cols = X.shape[1] # number of predictors
print("Number of features:", n_cols)

Number of features: 8


<a id="item1"></a>


<a id='item32'></a>


## Build a Neural Network


Let's define a function that defines our neural network model for us so that we can conveniently call it to create our model.


In [11]:
# define regression model
def create_neural_network():
    # create model
    model = Sequential()
    model.add(Dense(10, activation='relu', input_shape=(n_cols,)))
    model.add(Dense(1))
    
    # compile model
    model.compile(optimizer='adam', loss='mean_squared_error')
    return model

The above function create a model that has one hidden layers, each of 10 hidden units.


<a id="item4"></a>


<a id='item34'></a>


## Train and Test the Network


Let's call the function now to create our model.


In [12]:
# build the model
model = create_neural_network()

Next, we will train and test the model at the same time using the *fit* method. We will leave out 30% of the data for validation and we will train the model for 100 epochs.


In [13]:
# fit the model
mean_sq_error = [] # Empty list of mean squared error

def split_and_fit_model(myModel):
    X_train, X_test, y_train, y_test = train_test_split( X, y, test_size=0.3, random_state=4)
    
    myModel.fit(X_train, y_train, validation_split=0.3, epochs=50, verbose=2)
    
    # evaluate the model
    scores = myModel.evaluate(X_test, y_test, verbose=0)
    print("Accuracy: {} \n Error: {}".format(scores, 100-scores*100))
    
    # Get the predicted values
    yhat = myModel.predict(X_test)
    mse_sk = metrics.mean_squared_error(y_test, yhat) # Calculate RMSE
    mean_sq_error.append(mse_sk)

# Step 4
for i in range(50):
    print("Iteration %s:\n" %str(i))
    split_and_fit_model(model)
    
print(len(mean_sq_error))
print("Mean:", np.mean(np.array(mean_sq_error)))
print("Standard Deviation :", np.std(np.array(mean_sq_error)))

Iteration 0:

Epoch 1/50
16/16 - 1s - loss: 3337.1255 - val_loss: 3719.6260 - 535ms/epoch - 33ms/step
Epoch 2/50
16/16 - 0s - loss: 2821.7224 - val_loss: 3208.6472 - 38ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 2450.3745 - val_loss: 2770.4177 - 36ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 2180.5139 - val_loss: 2424.2852 - 37ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 1913.3873 - val_loss: 2166.1641 - 35ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 1691.3905 - val_loss: 1894.3970 - 36ms/epoch - 2ms/step
Epoch 7/50
16/16 - 0s - loss: 1533.2555 - val_loss: 1712.7974 - 38ms/epoch - 2ms/step
Epoch 8/50
16/16 - 0s - loss: 1363.9476 - val_loss: 1516.8876 - 38ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 1248.8595 - val_loss: 1359.4110 - 34ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 1132.9487 - val_loss: 1220.0590 - 38ms/epoch - 2ms/step
Epoch 11/50
16/16 - 0s - loss: 1046.6863 - val_loss: 1138.6962 - 34ms/epoch - 2ms/step
Epoch 12/50
16/16 - 0s - loss: 944.5

Epoch 46/50
16/16 - 0s - loss: 110.7398 - val_loss: 103.3473 - 36ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 113.5203 - val_loss: 140.4639 - 36ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 114.6563 - val_loss: 123.1609 - 61ms/epoch - 4ms/step
Epoch 49/50
16/16 - 0s - loss: 100.6535 - val_loss: 104.2542 - 37ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 100.9079 - val_loss: 102.0563 - 38ms/epoch - 2ms/step
Accuracy: 118.41254425048828 
 Error: -11741.254425048828
10/10 [==============================] - 0s 889us/step
Iteration 2:

Epoch 1/50
16/16 - 0s - loss: 101.0273 - val_loss: 102.5650 - 72ms/epoch - 4ms/step
Epoch 2/50
16/16 - 0s - loss: 100.8407 - val_loss: 100.4947 - 37ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 101.2690 - val_loss: 101.3590 - 37ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 100.4489 - val_loss: 101.2364 - 35ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 105.9180 - val_loss: 108.6815 - 35ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss

Epoch 40/50
16/16 - 0s - loss: 98.7783 - val_loss: 101.5569 - 37ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 99.7888 - val_loss: 102.3857 - 34ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 99.9691 - val_loss: 101.9500 - 35ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 102.2114 - val_loss: 114.2591 - 35ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 101.2271 - val_loss: 108.7986 - 34ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 103.9770 - val_loss: 100.1469 - 34ms/epoch - 2ms/step
Epoch 46/50
16/16 - 0s - loss: 101.3279 - val_loss: 99.6523 - 35ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 101.6854 - val_loss: 110.5789 - 36ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 100.4910 - val_loss: 102.0385 - 34ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 99.7535 - val_loss: 100.0281 - 36ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 103.3018 - val_loss: 103.7282 - 36ms/epoch - 2ms/step
Accuracy: 121.3896484375 
 Error: -12038.96484375
10/10 [=============

16/16 - 0s - loss: 103.7665 - val_loss: 114.6777 - 36ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 105.6483 - val_loss: 103.4290 - 33ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 99.3624 - val_loss: 100.4781 - 33ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 100.7102 - val_loss: 100.3582 - 36ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 99.6850 - val_loss: 104.3624 - 36ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 99.2002 - val_loss: 104.0911 - 36ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 100.3334 - val_loss: 101.1373 - 37ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 100.0462 - val_loss: 100.9153 - 34ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 100.1133 - val_loss: 101.6190 - 32ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 99.9122 - val_loss: 106.4131 - 36ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 107.4814 - val_loss: 110.2523 - 37ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 114.0254 - val_loss: 110.1262 - 37ms/epoch - 2ms/s

16/16 - 0s - loss: 101.1833 - val_loss: 109.3366 - 34ms/epoch - 2ms/step
Epoch 29/50
16/16 - 0s - loss: 104.0698 - val_loss: 105.0005 - 34ms/epoch - 2ms/step
Epoch 30/50
16/16 - 0s - loss: 106.7549 - val_loss: 100.5259 - 33ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 102.4576 - val_loss: 108.1667 - 32ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 102.8076 - val_loss: 104.6401 - 35ms/epoch - 2ms/step
Epoch 33/50
16/16 - 0s - loss: 103.2277 - val_loss: 118.7365 - 38ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 101.3454 - val_loss: 101.7321 - 37ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 98.1914 - val_loss: 103.0245 - 37ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 101.9641 - val_loss: 110.6148 - 38ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 105.2893 - val_loss: 100.3820 - 36ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 103.8106 - val_loss: 102.2393 - 36ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 101.1528 - val_loss: 99.9952 - 35ms/epoch - 2ms

Epoch 23/50
16/16 - 0s - loss: 106.2008 - val_loss: 120.1594 - 33ms/epoch - 2ms/step
Epoch 24/50
16/16 - 0s - loss: 103.5685 - val_loss: 105.0122 - 37ms/epoch - 2ms/step
Epoch 25/50
16/16 - 0s - loss: 100.0001 - val_loss: 101.3753 - 37ms/epoch - 2ms/step
Epoch 26/50
16/16 - 0s - loss: 99.4902 - val_loss: 99.7847 - 35ms/epoch - 2ms/step
Epoch 27/50
16/16 - 0s - loss: 103.0390 - val_loss: 110.5035 - 35ms/epoch - 2ms/step
Epoch 28/50
16/16 - 0s - loss: 103.8251 - val_loss: 101.6036 - 35ms/epoch - 2ms/step
Epoch 29/50
16/16 - 0s - loss: 99.0769 - val_loss: 102.3617 - 37ms/epoch - 2ms/step
Epoch 30/50
16/16 - 0s - loss: 103.1772 - val_loss: 109.9853 - 36ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 102.2058 - val_loss: 100.1260 - 37ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 101.0174 - val_loss: 101.2009 - 34ms/epoch - 2ms/step
Epoch 33/50
16/16 - 0s - loss: 98.1140 - val_loss: 99.8676 - 37ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 98.0920 - val_loss: 100.5047 - 37ms/epo

Epoch 18/50
16/16 - 0s - loss: 100.1066 - val_loss: 99.2834 - 37ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 98.1760 - val_loss: 102.5065 - 36ms/epoch - 2ms/step
Epoch 20/50
16/16 - 0s - loss: 99.8627 - val_loss: 120.2509 - 38ms/epoch - 2ms/step
Epoch 21/50
16/16 - 0s - loss: 103.8775 - val_loss: 108.4265 - 37ms/epoch - 2ms/step
Epoch 22/50
16/16 - 0s - loss: 102.1885 - val_loss: 104.0180 - 36ms/epoch - 2ms/step
Epoch 23/50
16/16 - 0s - loss: 101.8577 - val_loss: 106.3495 - 34ms/epoch - 2ms/step
Epoch 24/50
16/16 - 0s - loss: 99.4805 - val_loss: 119.0056 - 35ms/epoch - 2ms/step
Epoch 25/50
16/16 - 0s - loss: 98.0103 - val_loss: 100.7966 - 35ms/epoch - 2ms/step
Epoch 26/50
16/16 - 0s - loss: 99.1883 - val_loss: 100.1189 - 35ms/epoch - 2ms/step
Epoch 27/50
16/16 - 0s - loss: 98.1743 - val_loss: 103.3779 - 38ms/epoch - 2ms/step
Epoch 28/50
16/16 - 0s - loss: 99.5260 - val_loss: 100.6491 - 37ms/epoch - 2ms/step
Epoch 29/50
16/16 - 0s - loss: 101.4923 - val_loss: 111.9601 - 35ms/epoch

Epoch 13/50
16/16 - 0s - loss: 98.6303 - val_loss: 102.1777 - 35ms/epoch - 2ms/step
Epoch 14/50
16/16 - 0s - loss: 99.8561 - val_loss: 102.1534 - 37ms/epoch - 2ms/step
Epoch 15/50
16/16 - 0s - loss: 100.5240 - val_loss: 99.1812 - 36ms/epoch - 2ms/step
Epoch 16/50
16/16 - 0s - loss: 97.5969 - val_loss: 99.4206 - 36ms/epoch - 2ms/step
Epoch 17/50
16/16 - 0s - loss: 100.5268 - val_loss: 111.9031 - 35ms/epoch - 2ms/step
Epoch 18/50
16/16 - 0s - loss: 100.4027 - val_loss: 99.6862 - 36ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 100.7101 - val_loss: 107.0912 - 37ms/epoch - 2ms/step
Epoch 20/50
16/16 - 0s - loss: 100.7411 - val_loss: 99.3324 - 37ms/epoch - 2ms/step
Epoch 21/50
16/16 - 0s - loss: 102.8150 - val_loss: 118.8812 - 35ms/epoch - 2ms/step
Epoch 22/50
16/16 - 0s - loss: 106.2869 - val_loss: 98.9270 - 35ms/epoch - 2ms/step
Epoch 23/50
16/16 - 0s - loss: 100.6111 - val_loss: 99.2792 - 35ms/epoch - 2ms/step
Epoch 24/50
16/16 - 0s - loss: 100.0261 - val_loss: 112.0761 - 35ms/epoch 

Epoch 8/50
16/16 - 0s - loss: 98.9146 - val_loss: 108.0281 - 38ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 101.4512 - val_loss: 105.3267 - 34ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 115.8788 - val_loss: 136.5873 - 35ms/epoch - 2ms/step
Epoch 11/50
16/16 - 0s - loss: 110.4849 - val_loss: 106.1564 - 36ms/epoch - 2ms/step
Epoch 12/50
16/16 - 0s - loss: 99.6379 - val_loss: 105.9454 - 34ms/epoch - 2ms/step
Epoch 13/50
16/16 - 0s - loss: 105.9983 - val_loss: 100.1805 - 31ms/epoch - 2ms/step
Epoch 14/50
16/16 - 0s - loss: 101.8631 - val_loss: 105.8635 - 35ms/epoch - 2ms/step
Epoch 15/50
16/16 - 0s - loss: 108.6335 - val_loss: 105.2876 - 35ms/epoch - 2ms/step
Epoch 16/50
16/16 - 0s - loss: 103.3107 - val_loss: 100.0642 - 34ms/epoch - 2ms/step
Epoch 17/50
16/16 - 0s - loss: 98.8627 - val_loss: 101.5163 - 37ms/epoch - 2ms/step
Epoch 18/50
16/16 - 0s - loss: 99.9364 - val_loss: 107.4244 - 34ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 99.1880 - val_loss: 98.8929 - 34ms/epoch

Epoch 3/50
16/16 - 0s - loss: 97.9659 - val_loss: 98.8269 - 36ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 101.5583 - val_loss: 102.5764 - 34ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 105.3738 - val_loss: 108.3441 - 36ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 99.3505 - val_loss: 101.7331 - 34ms/epoch - 2ms/step
Epoch 7/50
16/16 - 0s - loss: 99.3324 - val_loss: 108.2368 - 37ms/epoch - 2ms/step
Epoch 8/50
16/16 - 0s - loss: 101.9063 - val_loss: 102.4506 - 34ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 102.4401 - val_loss: 99.0240 - 34ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 101.1141 - val_loss: 116.7763 - 35ms/epoch - 2ms/step
Epoch 11/50
16/16 - 0s - loss: 105.5550 - val_loss: 100.4043 - 36ms/epoch - 2ms/step
Epoch 12/50
16/16 - 0s - loss: 101.7370 - val_loss: 103.6468 - 36ms/epoch - 2ms/step
Epoch 13/50
16/16 - 0s - loss: 103.5260 - val_loss: 99.6899 - 35ms/epoch - 2ms/step
Epoch 14/50
16/16 - 0s - loss: 98.7085 - val_loss: 99.8885 - 38ms/epoch - 2ms/

Epoch 49/50
16/16 - 0s - loss: 101.7223 - val_loss: 98.6440 - 33ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 99.1325 - val_loss: 100.6891 - 37ms/epoch - 2ms/step
Accuracy: 116.52567291259766 
 Error: -11552.567291259766
10/10 [==============================] - 0s 868us/step
Iteration 19:

Epoch 1/50
16/16 - 0s - loss: 98.6210 - val_loss: 109.9237 - 71ms/epoch - 4ms/step
Epoch 2/50
16/16 - 0s - loss: 99.4369 - val_loss: 107.0495 - 35ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 98.8238 - val_loss: 103.3486 - 34ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 98.9393 - val_loss: 99.8469 - 37ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 97.6503 - val_loss: 101.8681 - 35ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 101.0401 - val_loss: 98.9880 - 35ms/epoch - 2ms/step
Epoch 7/50
16/16 - 0s - loss: 99.9130 - val_loss: 99.1864 - 38ms/epoch - 2ms/step
Epoch 8/50
16/16 - 0s - loss: 100.5121 - val_loss: 102.8249 - 36ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 99.2842 - v

Epoch 44/50
16/16 - 0s - loss: 102.1572 - val_loss: 111.6052 - 32ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 104.1420 - val_loss: 102.1224 - 33ms/epoch - 2ms/step
Epoch 46/50
16/16 - 0s - loss: 98.1554 - val_loss: 105.8357 - 34ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 100.9044 - val_loss: 99.7116 - 35ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 100.1889 - val_loss: 105.1125 - 36ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 99.5341 - val_loss: 98.8065 - 38ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 103.0246 - val_loss: 99.5226 - 37ms/epoch - 2ms/step
Accuracy: 115.4933090209961 
 Error: -11449.33090209961
10/10 [==============================] - 0s 891us/step
Iteration 21:

Epoch 1/50
16/16 - 0s - loss: 98.0190 - val_loss: 99.1724 - 70ms/epoch - 4ms/step
Epoch 2/50
16/16 - 0s - loss: 108.1774 - val_loss: 116.1073 - 35ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 104.0889 - val_loss: 99.3927 - 34ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 98.00

Epoch 39/50
16/16 - 0s - loss: 100.9648 - val_loss: 102.6749 - 33ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 103.1168 - val_loss: 101.0093 - 34ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 104.3867 - val_loss: 98.7389 - 34ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 106.8979 - val_loss: 108.7011 - 34ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 101.4687 - val_loss: 112.3047 - 36ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 99.7393 - val_loss: 99.6240 - 34ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 102.4943 - val_loss: 99.4578 - 34ms/epoch - 2ms/step
Epoch 46/50
16/16 - 0s - loss: 98.1726 - val_loss: 107.0948 - 33ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 100.7033 - val_loss: 114.5359 - 34ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 101.7381 - val_loss: 104.0387 - 38ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 98.4791 - val_loss: 101.4167 - 36ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 98.4747 - val_loss: 100.4321 - 35ms/epoc

Epoch 34/50
16/16 - 0s - loss: 100.4746 - val_loss: 108.8213 - 35ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 97.8943 - val_loss: 99.6469 - 34ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 103.5312 - val_loss: 99.7577 - 35ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 97.5116 - val_loss: 102.8941 - 34ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 105.6188 - val_loss: 125.8508 - 36ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 107.0029 - val_loss: 108.0519 - 35ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 106.0026 - val_loss: 99.6274 - 34ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 97.4954 - val_loss: 99.0061 - 35ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 97.7947 - val_loss: 100.3911 - 37ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 101.5937 - val_loss: 102.1770 - 37ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 99.0402 - val_loss: 105.6030 - 35ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 99.6325 - val_loss: 108.3440 - 34ms/epoch -

Epoch 29/50
16/16 - 0s - loss: 98.0077 - val_loss: 106.0038 - 37ms/epoch - 2ms/step
Epoch 30/50
16/16 - 0s - loss: 101.8716 - val_loss: 98.7395 - 38ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 104.5520 - val_loss: 98.4948 - 35ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 103.2501 - val_loss: 112.2404 - 32ms/epoch - 2ms/step
Epoch 33/50
16/16 - 0s - loss: 100.7230 - val_loss: 98.7711 - 34ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 98.2099 - val_loss: 98.8928 - 35ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 97.8363 - val_loss: 99.7974 - 35ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 99.2703 - val_loss: 101.6521 - 34ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 97.9628 - val_loss: 102.1941 - 34ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 101.9814 - val_loss: 100.5390 - 34ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 99.3396 - val_loss: 99.5571 - 35ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 98.2644 - val_loss: 102.1704 - 35ms/epoch - 2m

Epoch 24/50
16/16 - 0s - loss: 106.7407 - val_loss: 116.9643 - 35ms/epoch - 2ms/step
Epoch 25/50
16/16 - 0s - loss: 113.8325 - val_loss: 113.6784 - 37ms/epoch - 2ms/step
Epoch 26/50
16/16 - 0s - loss: 102.8717 - val_loss: 99.2180 - 35ms/epoch - 2ms/step
Epoch 27/50
16/16 - 0s - loss: 98.3539 - val_loss: 100.3163 - 37ms/epoch - 2ms/step
Epoch 28/50
16/16 - 0s - loss: 98.6250 - val_loss: 106.5205 - 34ms/epoch - 2ms/step
Epoch 29/50
16/16 - 0s - loss: 102.4105 - val_loss: 99.6353 - 34ms/epoch - 2ms/step
Epoch 30/50
16/16 - 0s - loss: 100.1749 - val_loss: 100.0553 - 37ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 101.7377 - val_loss: 99.4854 - 37ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 98.5769 - val_loss: 98.8512 - 34ms/epoch - 2ms/step
Epoch 33/50
16/16 - 0s - loss: 105.6157 - val_loss: 100.9696 - 35ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 102.0421 - val_loss: 98.7725 - 44ms/epoch - 3ms/step
Epoch 35/50
16/16 - 0s - loss: 98.2322 - val_loss: 103.1688 - 35ms/epoch 

Epoch 19/50
16/16 - 0s - loss: 100.0624 - val_loss: 106.7505 - 35ms/epoch - 2ms/step
Epoch 20/50
16/16 - 0s - loss: 99.1283 - val_loss: 100.3390 - 36ms/epoch - 2ms/step
Epoch 21/50
16/16 - 0s - loss: 100.9409 - val_loss: 120.6389 - 34ms/epoch - 2ms/step
Epoch 22/50
16/16 - 0s - loss: 102.0609 - val_loss: 99.2229 - 36ms/epoch - 2ms/step
Epoch 23/50
16/16 - 0s - loss: 98.9220 - val_loss: 99.4835 - 35ms/epoch - 2ms/step
Epoch 24/50
16/16 - 0s - loss: 98.4131 - val_loss: 100.5912 - 35ms/epoch - 2ms/step
Epoch 25/50
16/16 - 0s - loss: 98.3357 - val_loss: 106.6291 - 37ms/epoch - 2ms/step
Epoch 26/50
16/16 - 0s - loss: 99.9745 - val_loss: 98.8152 - 36ms/epoch - 2ms/step
Epoch 27/50
16/16 - 0s - loss: 97.4955 - val_loss: 106.1768 - 37ms/epoch - 2ms/step
Epoch 28/50
16/16 - 0s - loss: 98.2729 - val_loss: 98.8693 - 35ms/epoch - 2ms/step
Epoch 29/50
16/16 - 0s - loss: 99.5805 - val_loss: 115.7249 - 36ms/epoch - 2ms/step
Epoch 30/50
16/16 - 0s - loss: 98.9273 - val_loss: 98.9457 - 36ms/epoch - 2ms

Epoch 14/50
16/16 - 0s - loss: 106.3801 - val_loss: 100.6125 - 54ms/epoch - 3ms/step
Epoch 15/50
16/16 - 0s - loss: 100.7858 - val_loss: 100.1161 - 36ms/epoch - 2ms/step
Epoch 16/50
16/16 - 0s - loss: 99.6151 - val_loss: 104.1345 - 36ms/epoch - 2ms/step
Epoch 17/50
16/16 - 0s - loss: 103.8401 - val_loss: 107.8093 - 36ms/epoch - 2ms/step
Epoch 18/50
16/16 - 0s - loss: 97.6592 - val_loss: 100.2016 - 39ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 98.3738 - val_loss: 103.7457 - 32ms/epoch - 2ms/step
Epoch 20/50
16/16 - 0s - loss: 105.8729 - val_loss: 110.5339 - 33ms/epoch - 2ms/step
Epoch 21/50
16/16 - 0s - loss: 106.9656 - val_loss: 100.1550 - 36ms/epoch - 2ms/step
Epoch 22/50
16/16 - 0s - loss: 109.3236 - val_loss: 109.5230 - 35ms/epoch - 2ms/step
Epoch 23/50
16/16 - 0s - loss: 102.5821 - val_loss: 99.1026 - 35ms/epoch - 2ms/step
Epoch 24/50
16/16 - 0s - loss: 100.9737 - val_loss: 99.4501 - 37ms/epoch - 2ms/step
Epoch 25/50
16/16 - 0s - loss: 99.6161 - val_loss: 99.0531 - 39ms/epoc

Epoch 9/50
16/16 - 0s - loss: 98.1770 - val_loss: 100.3010 - 34ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 98.6427 - val_loss: 101.0404 - 33ms/epoch - 2ms/step
Epoch 11/50
16/16 - 0s - loss: 99.1240 - val_loss: 99.7277 - 34ms/epoch - 2ms/step
Epoch 12/50
16/16 - 0s - loss: 100.4649 - val_loss: 101.1062 - 34ms/epoch - 2ms/step
Epoch 13/50
16/16 - 0s - loss: 99.1544 - val_loss: 98.8955 - 37ms/epoch - 2ms/step
Epoch 14/50
16/16 - 0s - loss: 97.6404 - val_loss: 114.1109 - 34ms/epoch - 2ms/step
Epoch 15/50
16/16 - 0s - loss: 100.6163 - val_loss: 101.6092 - 34ms/epoch - 2ms/step
Epoch 16/50
16/16 - 0s - loss: 98.4403 - val_loss: 99.4827 - 35ms/epoch - 2ms/step
Epoch 17/50
16/16 - 0s - loss: 104.7467 - val_loss: 100.5192 - 35ms/epoch - 2ms/step
Epoch 18/50
16/16 - 0s - loss: 98.8498 - val_loss: 104.9856 - 35ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 100.5454 - val_loss: 101.3163 - 36ms/epoch - 2ms/step
Epoch 20/50
16/16 - 0s - loss: 103.3431 - val_loss: 118.1540 - 37ms/epoch - 

Epoch 4/50
16/16 - 0s - loss: 97.9505 - val_loss: 101.6413 - 35ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 99.3600 - val_loss: 99.6696 - 36ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 100.5616 - val_loss: 103.4509 - 34ms/epoch - 2ms/step
Epoch 7/50
16/16 - 0s - loss: 102.2277 - val_loss: 99.2479 - 37ms/epoch - 2ms/step
Epoch 8/50
16/16 - 0s - loss: 98.4269 - val_loss: 99.1427 - 38ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 99.3810 - val_loss: 100.5490 - 35ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 98.6096 - val_loss: 100.3601 - 35ms/epoch - 2ms/step
Epoch 11/50
16/16 - 0s - loss: 99.0415 - val_loss: 98.7297 - 35ms/epoch - 2ms/step
Epoch 12/50
16/16 - 0s - loss: 99.5076 - val_loss: 106.4662 - 36ms/epoch - 2ms/step
Epoch 13/50
16/16 - 0s - loss: 99.7212 - val_loss: 104.6279 - 39ms/epoch - 2ms/step
Epoch 14/50
16/16 - 0s - loss: 99.9936 - val_loss: 99.1990 - 39ms/epoch - 2ms/step
Epoch 15/50
16/16 - 0s - loss: 102.9712 - val_loss: 99.6473 - 38ms/epoch - 2ms/step
E

16/16 - 0s - loss: 94.6331 - val_loss: 95.5222 - 36ms/epoch - 2ms/step
Accuracy: 109.5893783569336 
 Error: -10858.93783569336
10/10 [==============================] - 0s 1ms/step
Iteration 38:

Epoch 1/50
16/16 - 0s - loss: 94.6907 - val_loss: 97.6106 - 69ms/epoch - 4ms/step
Epoch 2/50
16/16 - 0s - loss: 94.5732 - val_loss: 108.3501 - 36ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 92.5640 - val_loss: 94.4012 - 36ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 93.4363 - val_loss: 94.4063 - 36ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 91.6438 - val_loss: 93.1991 - 38ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 90.7921 - val_loss: 92.6882 - 33ms/epoch - 2ms/step
Epoch 7/50
16/16 - 0s - loss: 89.7733 - val_loss: 99.4668 - 36ms/epoch - 2ms/step
Epoch 8/50
16/16 - 0s - loss: 92.2174 - val_loss: 94.1236 - 36ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 88.3966 - val_loss: 93.0070 - 35ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 88.1104 - val_loss: 98.2726 - 35ms/

16/16 - 0s - loss: 54.3752 - val_loss: 49.3735 - 34ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 56.8040 - val_loss: 71.5316 - 34ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 56.7338 - val_loss: 50.6648 - 36ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 52.3669 - val_loss: 49.4973 - 37ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 55.9311 - val_loss: 52.4104 - 35ms/epoch - 2ms/step
Accuracy: 54.84310531616211 
 Error: -5384.310531616211
10/10 [==============================] - 0s 998us/step
Iteration 40:

Epoch 1/50
16/16 - 0s - loss: 56.9519 - val_loss: 58.0197 - 72ms/epoch - 4ms/step
Epoch 2/50
16/16 - 0s - loss: 52.1993 - val_loss: 49.4174 - 36ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 53.9782 - val_loss: 52.1644 - 38ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 52.4586 - val_loss: 51.5205 - 35ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 52.8975 - val_loss: 50.1640 - 36ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 52.3299 - val_loss: 50.3189 - 3

16/16 - 0s - loss: 51.5771 - val_loss: 63.3485 - 35ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 51.6141 - val_loss: 50.3992 - 39ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 50.7473 - val_loss: 51.5711 - 38ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 53.8721 - val_loss: 46.8466 - 33ms/epoch - 2ms/step
Epoch 46/50
16/16 - 0s - loss: 49.8886 - val_loss: 47.0244 - 34ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 51.2033 - val_loss: 46.5647 - 32ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 51.0342 - val_loss: 46.2068 - 36ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 50.2528 - val_loss: 46.4787 - 33ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 49.9964 - val_loss: 47.3850 - 34ms/epoch - 2ms/step
Accuracy: 51.01714324951172 
 Error: -5001.714324951172
10/10 [==============================] - 0s 1ms/step
Iteration 42:

Epoch 1/50
16/16 - 0s - loss: 50.6982 - val_loss: 47.8512 - 81ms/epoch - 5ms/step
Epoch 2/50
16/16 - 0s - loss: 50.4678 - val_loss: 46.7319 -

16/16 - 0s - loss: 50.5423 - val_loss: 45.8188 - 34ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 50.6787 - val_loss: 57.2737 - 34ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 50.1843 - val_loss: 48.6147 - 37ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 51.9683 - val_loss: 51.4662 - 35ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 53.7540 - val_loss: 61.3532 - 37ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 52.4411 - val_loss: 45.2720 - 34ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 48.9367 - val_loss: 49.4558 - 33ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 48.7964 - val_loss: 45.6177 - 32ms/epoch - 2ms/step
Epoch 46/50
16/16 - 0s - loss: 49.1991 - val_loss: 45.8925 - 33ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 49.5793 - val_loss: 45.5590 - 36ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 50.1795 - val_loss: 45.5683 - 33ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 49.9836 - val_loss: 45.4102 - 34ms/epoch - 2ms/step
Epoch 50/50
16/1

16/16 - 0s - loss: 49.2925 - val_loss: 49.0840 - 34ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 50.1448 - val_loss: 47.2112 - 34ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 50.5626 - val_loss: 53.2323 - 32ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 50.8827 - val_loss: 53.7127 - 33ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 49.6768 - val_loss: 45.7621 - 35ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 51.9572 - val_loss: 48.1108 - 35ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 55.0750 - val_loss: 45.4277 - 38ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 51.7264 - val_loss: 46.2156 - 34ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 50.2124 - val_loss: 63.0329 - 37ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 54.2169 - val_loss: 59.4012 - 38ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 59.6144 - val_loss: 54.3490 - 33ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 51.7525 - val_loss: 47.3551 - 32ms/epoch - 2ms/step
Epoch 46/50
16/1

16/16 - 0s - loss: 48.8783 - val_loss: 45.3566 - 33ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 48.5363 - val_loss: 51.1563 - 33ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 48.2529 - val_loss: 57.1436 - 34ms/epoch - 2ms/step
Epoch 33/50
16/16 - 0s - loss: 53.3461 - val_loss: 45.7494 - 36ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 49.8354 - val_loss: 53.4726 - 36ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 49.3882 - val_loss: 45.2886 - 35ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 48.5990 - val_loss: 46.7475 - 35ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 48.2803 - val_loss: 45.0662 - 35ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 49.5348 - val_loss: 48.2023 - 36ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 49.0529 - val_loss: 47.9347 - 37ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 49.4225 - val_loss: 48.5967 - 33ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 48.4338 - val_loss: 45.1448 - 32ms/epoch - 2ms/step
Epoch 42/50
16/1

16/16 - 0s - loss: 49.4967 - val_loss: 45.8553 - 34ms/epoch - 2ms/step
Epoch 27/50
16/16 - 0s - loss: 50.2205 - val_loss: 44.8242 - 32ms/epoch - 2ms/step
Epoch 28/50
16/16 - 0s - loss: 49.9268 - val_loss: 57.0876 - 36ms/epoch - 2ms/step
Epoch 29/50
16/16 - 0s - loss: 49.3968 - val_loss: 49.9679 - 34ms/epoch - 2ms/step
Epoch 30/50
16/16 - 0s - loss: 48.3475 - val_loss: 46.2993 - 34ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 49.2514 - val_loss: 45.5196 - 35ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 48.2616 - val_loss: 45.1926 - 34ms/epoch - 2ms/step
Epoch 33/50
16/16 - 0s - loss: 49.2572 - val_loss: 48.2619 - 33ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 48.8846 - val_loss: 58.1362 - 33ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 49.5268 - val_loss: 45.9316 - 35ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 48.1357 - val_loss: 46.9609 - 34ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 49.6966 - val_loss: 45.1720 - 35ms/epoch - 2ms/step
Epoch 38/50
16/1